### PR TITLE
1591 Add paging to search

### PR DESCRIPTION
**Pull request by Mergoth**

### CODE REVIEW

**Review by mrfirelord — APPROVED** [review]

File: classificators/src/main/java/ru/okpdmarket/controller/ClassificatorController.java [hunk 1]
```diff
@@ -50,7 +51,10 @@ public ClassificatorItem getItem(@PathVariable(value = "classificatorCode") Stri
     }
 
     @RequestMapping(value = "/{code}/search", method = RequestMethod.GET)
-    public List<ClassificatorItem> search(@PathVariable(value = "code") String classificatorCode, @RequestParam String query) {
-        return searchService.searchByClassificator(classificatorCode, query);
+    public SearchResult search(@PathVariable(value = "code") String classificatorCode,
+                               @RequestParam(value = "query") String query,
+                               @RequestParam(value = "take", defaultValue = "${application.search.defaultTake}") Integer take,
+                               @RequestParam(value = "offset", defaultValue = "0") Integer offset) {
```
Comment (Vladlis) [inline classificators/src/main/java/ru/okpdmarket/controller/ClassificatorController.java:57]: Shouldn't this default value also be in properties?

File: classificators/src/main/java/ru/okpdmarket/service/impl/LuceneSearchServiceImpl.java [hunk 2]
```diff
@@ -97,24 +99,30 @@ public void indexClassificator(Classificator classificator) {
             finalQuery.add(query, BooleanClause.Occur.MUST);
             finalQuery.add(clsFilter, BooleanClause.Occur.MUST);
 
+            TopScoreDocCollector collector = TopScoreDocCollector.create(9999);
```
Comment (Vladlis) [inline classificators/src/main/java/ru/okpdmarket/service/impl/LuceneSearchServiceImpl.java]: 9999 - magic number

**Review by Vladlis — DISMISSED** [review]

File: classificators/src/main/java/ru/okpdmarket/controller/ClassificatorController.java (hunk 1 above)
Comment (Mergoth) [reply classificators/src/main/java/ru/okpdmarket/controller/ClassificatorController.java:57]: Nope, what for? Can you imagine any usecase?

File: classificators/src/main/java/ru/okpdmarket/service/impl/LuceneSearchServiceImpl.java (hunk 2 above)
Comment (Mergoth) [reply classificators/src/main/java/ru/okpdmarket/service/impl/LuceneSearchServiceImpl.java]: fair enough, will fix

File: classificators/src/main/java/ru/okpdmarket/controller/ClassificatorController.java (hunk 1 above)
Comment (Vladlis) [reply classificators/src/main/java/ru/okpdmarket/controller/ClassificatorController.java:57]: ok, I can't

**Comment by ohhaithere** [comment]
moved magic number to a constant

File: classificators/src/main/java/ru/okpdmarket/service/impl/LuceneSearchServiceImpl.java (hunk 2 above)
Comment (ohhaithere) [reply classificators/src/main/java/ru/okpdmarket/service/impl/LuceneSearchServiceImpl.java]: fixed

File: classificators/src/main/java/ru/okpdmarket/service/impl/LuceneSearchServiceImpl.java [hunk 3]
```diff
@@ -43,6 +43,7 @@
     private static final String CLS_ID = "classificator_code";
     private static final String ITEM_ID = "item_code";
     private static final Logger log = LoggerFactory.getLogger(SearchService.class);
+    private static final Integer NUMBER_OF_HITS = 9999;
```
Comment (Mergoth) [inline classificators/src/main/java/ru/okpdmarket/service/impl/LuceneSearchServiceImpl.java]: Wouldn't it be better to make it property? And name something like "..search.hits.max"?

Comment (ohhaithere) [reply classificators/src/main/java/ru/okpdmarket/service/impl/LuceneSearchServiceImpl.java]: I considered using a property, but as far as I understand this value won't change so I used constant instead. If I'm wrong, then yeah, gotta change it to prop

**Review by ohhaithere — DISMISSED** [review]

**Review by ohhaithere — APPROVED** [review]

**Pull request merged**